features/step_definitions/bed_verification_steps.rb

```diff
@@ -24,6 +24,41 @@
   )
 end
 
+# Mocking Sequencescape::Api::V2::Labware.where(barcode: barcodes)
+Given(
+  /^I can retrieve the labware with barcodes "([^"]*)" and lifespans "([^"]*)" and ages "([^"]*)" and existence "([^"]*)"$/
+) do |barcodes, lifespans, ages, existence|
+  barcode_list = barcodes.split(',').map(&:strip).reject(&:blank?)
+  lifespan_list =
+    lifespans
+      .split(',')
+      .reject(&:blank?)
+      .map do |lifespan|
+        lifespan.strip!
+        lifespan == 'nil' ? nil : lifespan.to_i
+      end
+  age_list = ages.split(',').reject(&:blank?).map(&:to_i)
+  exists_list = existence.split(',').reject(&:blank?).map { |e| e.strip == 'true' }
+
+  labware_list = []
+  barcode_list.each_with_index do |barcode, index|
+    next unless exists_list[index]
+
+    purpose = Sequencescape::Api::V2::Purpose.new
+    allow(purpose).to receive(:lifespan).and_return(lifespan_list[index])
+    allow(purpose).to receive(:name).and_return('immortal')
+
+    labware = Sequencescape::Api::V2::Labware.new
+    allow(labware).to receive(:purpose).and_return(purpose)
+    allow(labware).to receive(:labware_barcode).and_return({ 'machine_barcode' => barcode.to_s })
+    allow(labware).to receive(:created_at).and_return(Time.zone.today - age_list[index])
+
+    labware_list << labware
+  end
+
+  allow(Sequencescape::Api::V2::Labware).to receive(:where).with(barcode: barcode_list).and_return(labware_list)
+end
```

Comment:
I had a bit of trouble with how best to do these tests & mocking - any suggestions to make it nicer welcome.

Reply:
Sorry I might not be much help - still trying to figure out where the expectation is done?!

Reply:
This bit is all just for mocking this API call: `Sequencescape::Api::V2::Labware.where(barcode: barcodes)`

The expectations are in a different part of the Cucumber tests - file above, lines like `Then I should see "<message>"` and `And I should be on the new audit page` - the implementation behind those steps will be in a different file.